src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -3117,49 +3119,77 @@ impl Coordinator {
                             MutationKind::Insert => diffs.push((row.to_owned(), Diff::ONE)),
                         }
                     }
+
+                    // Sum of all the rows' byte size, for checking if we go
+                    // above the max_result_size threshold.
+                    let mut byte_size: u64 = 0;
                     for (row, diff) in &diffs {
+                        byte_size = byte_size.saturating_add(u64::cast_from(row.byte_len()));
                         if diff.is_positive() {
                             for (idx, datum) in row.iter().enumerate() {
                                 desc.constraints_met(idx, &datum)?;
                             }
                         }
                     }
-                    Ok(diffs)
+                    Ok((diffs, byte_size))
                 };
+
             let diffs = match peek_response {
-                ExecuteResponse::SendingRows { future: batch, .. } => {
-                    // TODO(jkosh44): This timeout should be removed;
-                    // we should instead periodically ensure clusters are
-                    // healthy and actively cancel any work waiting on unhealthy
-                    // clusters.
-                    match tokio::time::timeout(timeout_dur, batch).await {
-                        Ok(res) => match res {
-                            PeekResponseUnary::Rows(rows) => make_diffs(rows),
-                            PeekResponseUnary::Canceled => Err(AdapterError::Canceled),
-                            PeekResponseUnary::Error(e) => {
-                                Err(AdapterError::Unstructured(anyhow!(e)))
-                            }
-                        },
-                        Err(_) => {
-                            // We timed out, so remove the pending peek. This is
-                            // best-effort and doesn't guarantee we won't
-                            // receive a response.
-                            // It is not an error for this timeout to occur after `internal_cmd_rx` has been dropped.
-                            let result = internal_cmd_tx.send(Message::CancelPendingPeeks {
-                                conn_id: ctx.session().conn_id().clone(),
-                            });
-                            if let Err(e) = result {
-                                warn!("internal_cmd_rx dropped before we could send: {:?}", e);
+                ExecuteResponse::SendingRowsStreaming {
+                    rows: mut rows_stream,
+                    ..
+                } => {
+                    let mut byte_size: u64 = 0;
+                    let mut diffs = Vec::new();
+                    let result = loop {
+                        match tokio::time::timeout(timeout_dur, rows_stream.next()).await {
```

Comment:
This is subtly different behavior. If the stream drip feeds you batches the timeout is reset per batch. This might be fine (especially considering the removed TODO, but just something to consider.

Reply:
Good to call out! But yep, I'm also not too concerned